src/distance.rs

```diff
@@ -1,7 +1,6 @@
 pub fn squared_euclidean(a: &[f64], b: &[f64]) -> f64 {
-    let mut dist = 0f64;
-    for i in 0..a.len() {
```

Comment:
Note that this will silently fail to do the right thing if `a.len() < b.len()` and panic if `b.len() < a.len()` (though both should not apply in practice).